### ISSUE TITLE
Packrat doesn't work on Windows with a project on a network drive using UNC paths

### ISSUE THREAD

**Issue by pete-thompson** [rstudio/packrat#438]
I know you can map a drive to solve this problem, but that seems rather clunky - plus I spent a good hour or so trying to figure out my problem before I realised it was caused by the UNC path, it would have been nice to have saved that time!

Steps:
Open an existing project in RStudio that is stored on a network folder. Use the UNC style of path when opening the project (e.g. \\server\\share\folder)
The project should have been created using a different version of R - hence Packrat attempts to create new folders to handle the correct platform/version. 

initialisation fails, with errors similar to:

> Initializing packrat project in directory:
> - "\\server/share/folder"
> Packrat mode off. Resetting library paths to:
> - "C:/Users/myusername/Documents/R/R-3.4.3/library"
> Error in packageVersion("packrat") : package ‘packrat’ not found
> In addition: Warning messages:
> 1: In read.dcf(file.path(p, "DESCRIPTION"), c("Package", "Version")) :
>   cannot open compressed file '\\server/share/folder/packrat/lib/x86_64-w64-mingw32/3.4.3/packrat/DESCRIPTION', probable reason 'No such file or directory'
> 2: In read.dcf(file.path(p, "DESCRIPTION"), c("Package", "Version")) :
>   cannot open compressed file '\\server/share/folder/packrat/lib/x86_64-w64-mingw32/3.4.3/packrat/DESCRIPTION', probable reason 'No such file or directory'
> 3: In read.dcf(file.path(p, "DESCRIPTION"), c("Package", "Version")) :
>   cannot open compressed file \\server/share/folder/packrat/lib/x86_64-w64-mingw32/3.4.3/packrat/DESCRIPTION', probable reason 'No such file or directory'
> 4: In read.dcf(file.path(p, "DESCRIPTION"), c("Package", "Version")) :
>   cannot open compressed file \\server/share/folder/packrat/lib/x86_64-w64-mingw32/3.4.3/packrat/DESCRIPTION', probable reason 'No such file or directory'
> 

I think the problem is that Packrat fails to create the platform specific folder - at least when I check, the folder doesn't exist.

Similar errors happen when you attempt to create a Packrat project on a UNC based folder.

**Comment by pete-thompson** [comment]
I found at least one of the problems. During the initialisation there is code that attempts to install Packrat using the R command line. The parameters to this command include the location of the tarball for Packrat, using a relative path (I obviously have a cached copy in this instance under my Packrat folder). Running R through the command line uses the Windows CMD executable, which refuses to use a UNC path as the current directory - so the parameters to R are incorrect because they're using a relative path that now makes no sense. I might try and create a forked version and a pull request to fix this if I can figure out how to set up a development version of packrat!

**Comment by pete-thompson** [comment]
OK - so I did a little more debugging, got things to launch the command line with the UNC path and realised that it fails - dug a little deeper and realised that R simply doesn't support UNC paths for libraries, so clearly I'm wasting everyone's time! Time to close this issue until someone decides it's worth making R work against UNC paths (i.e. never!)

**Comment by kevinushey** [comment]
Yup :-) Unfortunately, paths on Windows in general with R can be a big headache!

**Comment by hack-r** [comment]
@pete-thompson  IMHO this is such a ginormous problem that (a) this should definitely be reopened and (b) R is severely limited relative to others for enterprise usage until proper virtual environments are working...

**Comment by pete-thompson** [comment]
The problem is with base R itself, hence there's nothing for the Packrat team to do, which is why I suggested they close my issue (I'm not a Packrat team member, just a normal user :)). I eventually tracked the problem down to a bug in R (bug number 17159 - https://bugs.r-project.org/bugzilla/show_bug.cgi?id=17159) which is apparently fixed. I haven't tested to see if the problem persists in the latest versions of R because I found a viable workaround.
